ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+  
+  * Akka Persistence is based upon event-sourcing, which means that the PersistentActor emits events that represent state changes to be applied. Snapshots may be used as an optimization to shorten recovery time.
+
+In both cases the persistence provider is configurable and users may choose to persist Actor state independently by other means as well. The biggest difference between snapshots and event-sourcing is that the event history of a PersistentActor contains valuable information that is lost when persisting only snapshots: each event has a well-defined meaning in the business domain and thus the event log retains its meaning as the internal representation of the Actor’s state changes over time while snapshots will become invalid and cannot be mined for additional insights later.
+
+#### Timers and Reminders
+
+  * Orleans provides transient timers that are local to a Grain activation and cease to exist when the Grain is passivated. It also provides reliable persistent reminders that will send an invocation to a Grain independently of whether it is currently active or not.
+  
+  * Akka includes only transient timers at this point, reliable reminders need to be implemented using third-party extensions like Akka Quartz.
+
+### Runtime Implementation
+
+#### Overview
+
+Both systems consist of similar core components: Orleans’ Execution corresponds to akka-actor in that it is responsible for the actual execution of Actors, Messaging corresponds to akka-remote in that it manages TCP connections between the cluster nodes, and Hosting corresponds to akka-cluster plus ClusterSharding. One difference is that akka-actor also allows purely local message sends that are optimized to local reference passing whereas Orleans invocations are always sent via the Messaging component that consults with the Hosting component as to where the recipient currently resides. Akka’s lack of indirection allows substantial performance optimizations in this regard.
+
+#### Distributed Directory
+
+  * Orleans uses a one-hop distributed hash table that maps GUIDs to their current activations’ locations by way of partitioning the GUID hash space onto the cluster nodes. Local caches are used to optimize away the network round-trip in 90% of the cases.
+  
+  * Akka uses a sharding approach where the hash space of the logical Actor names is partitioned into shards of equal size and a ClusterSingleton coordinates the placement of these shards onto cluster nodes.
+
+While Orleans’ Hosting has the ability to relocate individual Actors, it must pay for this ability by a hash table that grows proportionally to the number of Grains used. Akka’s hash-based approach in contrast distributes the full location information (on an as-needed basis) because its size is bounded by the configurable number of shards, which allows it to scale further than Orleans in the number of Actors that are running.
```

Comment:
This is a small implementation detail. We can also group actors by ids and map the whole group together (e.g., map each 100 actors via hashing into one directory entry). This will allow to overcome any potential directory size limitations (not that we have seen any such practical limitations, RAM is cheap) at the expense of  a bit less flexible placement. And of course having the directory being distributed eliminates the singleton part of the ClusterSingleton (better load spreading, more resilient, not a single point of failure).